pkg/test/framework/components/namespace/kube.go

```diff
@@ -84,7 +83,7 @@ func claimKube(ctx resource.Context, name string) (Instance, error) {
 }
 
 // NewNamespace allocates a new testing namespace.
-func newKube(ctx resource.Context, prefix string, inject bool) (Instance, error) {
+func newKube(ctx resource.Context, prefix string, inject bool, labels map[string]string) (Instance, error) {
```

Comment:
labels are an implementation detail ... I'd rather have explicit configuration for this. How about replace all the parameters with a config object.  Something like:

```go
type Config struct {
  Prefix string,
  Inject bool
  CustomSIdecarInjectorNamespace namespace.Instance
}
```

Reply:
+1
another reason to collapse into struct is it's hard to remember the order and meaning of all arguments.

Reply:
@nmittler I changed the way namespace test component configuration works a bit if you want to take another look

What I propose is that we have separate configs for each control-plane component that could potentially be configured through namespace labels, like so:

```golang
// Config contains configuration information about the namespace instance
// including naming information, sidecar injection config, and other component-specific labels
// note: each component that adds additional custom k8s labels for control should have a field here
type Config struct {
	Prefix        string
	SidecarConfig SidecarConfig
	CitadelConfig CitadelConfig
}

 // CitadelConfig contains the values for configuration of citadel-specific namespace labels
// if not included in config, namespace will have none of these labels
type CitadelConfig struct {
	ManagedLabel  string
	OverrideLabel string
}

 // SidecarConfig contains the values for configuration of sidecar-specific namespace labels
// if not included in config, namespace will have none of these labels
type SidecarConfig struct {
	Inject                  bool
	CustomInjectorNamespace string
}
```

This'll keep the config manageable, and extensible other component-specific namespace labels need testing.